docs: copy types into docs

cpb8010 opened the pull request:
Replacing unrelated interface information with typescript types. Also changes in copy to better reflect the standard compliance.

## Additional context

Opening a PR instead of pushing commits directly to another branch

keriat commented:
:tada: This PR is included in version 2.18.0 :tada:

The release is available on [GitHub release](https://github.com/matter-labs/zksync-docs/releases/tag/2.18.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

cpb8010 committed:
docs: copy types into docs

Replacing unrelated interface information with typescript types.
Also changes in copy to better reflect the standard compliance.
